File: Backend/Domain/NewsManager.cs
using System.Collections.Generic;
using Backend.Data.Database;
using Backend.Data.Entities;
using Backend.Data.Repositories;

namespace Backend.Domain
{
    public class NewsManager
    {
        private readonly DatabaseContext _databaseContext;
        private readonly NewsRepository _newsRepository;

        public NewsManager(DatabaseContext databaseContext, NewsRepository newsRepository)
        {
            _databaseContext = databaseContext;
            _newsRepository = newsRepository;
        }
        
        public IEnumerable<News> GetNewsAsyns()
        {
            return _newsRepository.GetNews();
        }
        
        public News GetNews(int id)
        {
            return _newsRepository.GetNews(id);
        }

        public News GetLastNews()
        {
            return _newsRepository.GetLastNews();
        }
        
        public void AddNews(News news)
        {
            _databaseContext.UseTransaction(transaction =>
                _newsRepository.AddNews(news, transaction)
            );
        }
    }
}

File: Android/app/src/main/java/com/khsm/app/domain/NewsManager.java
package com.khsm.app.domain;

import android.content.Context;
import android.support.annotation.NonNull;

import com.khsm.app.data.api.Api;
import com.khsm.app.data.api.ApiFactory;
import com.khsm.app.data.entities.Discipline;
import com.khsm.app.data.entities.News;

import java.util.List;

import io.reactivex.Single;

public class NewsManager {
    private final Api api;

    public NewsManager(@NonNull Context context) { api = ApiFactory.createApi(context); }

    public Single<List<News>> getNews() { return api.getNews(); }

    public Single<News> getLastNews() {
        return api.getLastNews();
    }

    public Single<News> addNews(News news) {
        return api.addNews(news);
    }
}
